web/src/lib/components/asset-viewer/asset-viewer.svelte

```diff
@@ -372,9 +372,18 @@
 
   const trashAsset = async () => {
     try {
+      const nextAsset = await assetStore?.getNextAssetId(asset.id);
+      const previousAsset = await assetStore?.getPreviousAssetId(asset.id);
+
       await api.assetApi.deleteAssets({ assetBulkDeleteDto: { ids: [asset.id] } });
 
-      await navigateAssetForward();
+      if (nextAsset) {
+        assetViewingStore.setAssetId(nextAsset);
+      } else if (previousAsset) {
+        assetViewingStore.setAssetId(previousAsset);
+      } else {
+        closeViewer();
+      }
```

Comment:
Is there a reason why we cannot put this logic inside of navigateAssetForward()?